data/logs_model/splunk_logs_model.py

```diff
@@ -0,0 +1,141 @@
+import json
+import logging
+
+from datetime import datetime
+
+from data import model
+from data.logs_model.interface import ActionLogsDataInterface
+from data.logs_model.logs_producer import LogProducerProxy, LogSendException
+from data.logs_model.logs_producer.splunk_logs_producer import SplunkLogsProducer
+from data.logs_model.shared import SharedModel
+from data.model import config
+from data.model.log import ACTIONS_ALLOWED_WITHOUT_AUDIT_LOGGING
+
+logger = logging.getLogger(__name__)
+
+
+class SplunkLogsModel(SharedModel, ActionLogsDataInterface):
+    """
+    SplunkLogsModel implements model for establishing connection and sending events to Splunk cluster
+    """
+
+    def __init__(self, producer, splunk_config, should_skip_logging=None):
+        self._should_skip_logging = should_skip_logging
+        self._logs_producer = LogProducerProxy()
+        if producer == "splunk":
+            self._logs_producer.initialize(SplunkLogsProducer(**splunk_config))
+        else:
+            raise Exception("Invalid log producer: %s" % producer)
+
+    def log_action(
+        self,
+        kind_name,
+        namespace_name=None,
+        performer=None,
+        ip=None,
+        metadata=None,
+        repository=None,
+        repository_name=None,
+        timestamp=None,
+        is_free_namespace=False,
+    ):
+
+        if self._should_skip_logging and self._should_skip_logging(
+            kind_name, namespace_name, is_free_namespace
+        ):
+            return
+
+        if repository_name is not None:
+            assert repository is None
+            assert namespace_name is not None
```

Comment:
Please don't do this. `assert` is for testing and debugging. If this code ever errors, customers will see an `AssertionError` in their logs, and that means very little to them. It's better to raise a generic `Exception` with a meaningful message instead.

Better even (although that would require a bigger refactor, which is out of the scope of this PR) would be change this function's arguments so that if one gives a repository name, then they _have_ to also give the namespace name. Having to do this kind of argument validation is a smell. But as I said, it's out of scope for you to handle to root cause, so just do the best you can with what you have.